core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncSigningStage.java

```diff
@@ -159,10 +159,6 @@ private static void updateAsyncRequestBodyInContexts(RequestExecutionContext con
     }
 
     private SdkHttpFullRequest toSdkHttpFullRequest(SignedRequest signedRequest) {
-        SdkHttpRequest request = signedRequest.request();
-        if (request instanceof SdkHttpFullRequest) {
```

Comment:
Note, `toSdkHttpFullRequest(AsyncSignedRequest signedRequest)` below still has this short-circuit, because for sync, we do not deal with `ContentStreamProvider`.